### PR TITLE
Fix continues events

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/oKpA8e88/4343-improvements-to-continuous-events

Reduce repair/compaction events severity.
Log continues events only into raw_events.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Test running - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity-2/79/

**Comment by enaydanov** [comment]
@bentsi @dkropachev `Severity.DEBUG` added by @fgelcer as a part of his work on https://github.com/scylladb/scylla-cluster-tests/pull/3299 

This severity should be used for grouping some events in reports.

File: sdcm/sct_events/file_logger.py [hunk 1]
```diff
@@ -79,22 +78,24 @@ def write_event(self, event: SctEvent) -> None:
...
+                with verbose_suppress("%s: failed to tee %s to %s", self, event, tee):
+                    tee(message)
+            message = message.encode("utf-8") + b"\n"
 
-        # Update events.log file (all events.)
-        with verbose_suppress("%s: failed to write %s to %s", self, event, self.events_log):
-            with self.events_log.open("ab+", buffering=0) as fobj:
-                fobj.write(message)
+            # Update events.log file (all events.)
+            if event.severity != Severity.DEBUG:
```
Comment (enaydanov) [inline sdcm/sct_events/file_logger.py:89]: Don't need this `if`

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: defaults/severities.yaml [hunk 2]
```diff
@@ -99,9 +99,9 @@ CassandraStressEvent: CRITICAL
 GeminiStressEvent: CRITICAL
 ScyllaServerStatusEvent: NORMAL
 BootstrapEvent: NORMAL
-RepairEvent: NORMAL
+RepairEvent: DEBUG
```
Comment (fgelcer) [inline defaults/severities.yaml:102]: why are these being set as DEBUG, and not like either NORMAL?

File: defaults/severities.yaml [hunk 3]
```diff
@@ -99,9 +99,9 @@ CassandraStressEvent: CRITICAL
...
 JMXServiceEvent: NORMAL
 InstanceStatusEvent.STARTUP: WARNING
 InstanceStatusEvent.REBOOT: WARNING
 InstanceStatusEvent.POWER_OFF: WARNING
-CompactionEvent: NORMAL
+CompactionEvent: DEBUG
```
Comment (fgelcer) [inline defaults/severities.yaml:107]: why are these being set as DEBUG, and not like either NORMAL?

**Comment by bentsi** [comment]
@roydahan @fgelcer The idea here is to remove compaction/repair events from all the events logs we have(events.log, normal.log, sct.log) and keep them in raw_events.log only for further usage in Parallel Timelines. It is needed to have full transparency of the processes that are happening in Scylla during the test run. Maybe choosing DEBUG level was not the best choice, but it removes all those events from the mentioned logs. 
- @dkropachev please handle the case for GrafanaAnnotator so we won't have those events added to the SCT dashboard
- let's merge this to ease the logging burden and think of a better solution

**Comment by fgelcer** [comment]
> @roydahan @fgelcer The idea here is to remove compaction/repair events from all the events logs we have(events.log, normal.log, sct.log) and keep them in raw_events.log only for further usage in Parallel Timelines. It is needed to have full transparency of the processes that are happening in Scylla during the test run. Maybe choosing DEBUG level was not the best choice, but it removes all those events from the mentioned logs.
> 
> * @dkropachev please handle the case for GrafanaAnnotator so we won't have those events added to the SCT dashboard
> * let's merge this to ease the logging burden and think of a better solution

@bentsi , the debug.log is being used for the reactor stalls, in the performance jobs... moving out from your way (events.log, normal.log, sct.log) to debug.log is not fixing the issue... also, the content of debug.log is right now being written to sct.log, so it will still be found in there (and please, don't fully remove the current debug.log content from sct.log, as they are being used by performance jobs)

File: sdcm/sct_events/file_logger.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/sct_events/file_logger.py]: fixed

File: defaults/severities.yaml (hunk 3 above)
Comment (dkropachev) [reply defaults/severities.yaml:107]: idea to do not throw them into the sct.log

File: defaults/severities.yaml (hunk 2 above)
Comment (fgelcer) [reply defaults/severities.yaml:102]: performance jobs need these in sct.log (it is used to decode the `Reactor stall` events

Comment (dkropachev) [reply defaults/severities.yaml:102]: @fgelcer , decoding events are picking up from db-node/messages.log, not from sct.log

**Comment by dkropachev** [comment]
Last run look super clean - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity-2/88/
Ready to be merged

File: defaults/severities.yaml (hunk 2 above)
Comment (fgelcer) [reply defaults/severities.yaml:102]: regular jobs decoding, yes, for performance jobs, it doesn't (yet) decode them automatically...
and right now sct.log is the file used to decode all the `Reactor stall` events, using `stall-analyser.py` tool inside seastar repo...

File: defaults/severities.yaml (hunk 3 above)
Comment (fgelcer) [reply defaults/severities.yaml:107]: AFAICT @roydahan wanted to get them removed completely, not hidden from the eye (and not where it doesn't belong to)

**Comment by bentsi** [comment]
Let's merge this to fix current issues and create a task to improve by adding a new attribute to events that will allow us to control whether to save events to logs

**Comment by ShlomiBalalis** [comment]
The issue occurs in 4.6 (as seen in $4209). Can you backport it to 4.6?

**Comment by roydahan** [comment]
Branch-4.6 is still in "fast forward" mode and is identical to master at this point.

**Pull request merged**